crowbar_framework/app/models/node_object.rb

```diff
@@ -1207,7 +1207,7 @@ def run_ssh_cmd(cmd)
     args = ["sudo", "-i", "-u", "root", "--", "timeout", "-k", "5s", "15s",
             "ssh", "-o", "ConnectTimeout=10",
             "root@#{@node.name}",
-            cmd.to_s
+            "\"#{cmd}\""
```

Comment:
No, it's the responsibility of the function to do its own escaping of double-quotes, because the requirement to escape them is hard-coupled to the decision to implement the function using double-quotes. Otherwise you are forcing the caller to make a privacy violation via assumptions about the function's internal implementation. Another approach would be to document the function's requirement that any double-quote characters are escaped before passing, which would be ugly but acceptable.

But it's pretty easy to fix this properly, so I'm not sure why you wouldn't want to do it.  Something like

    %("#{s.gsub('"', '\\"')}")

although maybe it needs to escape other characters too, like backslash.